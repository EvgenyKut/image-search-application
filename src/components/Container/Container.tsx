import React from 'react';
import styles from './container.module.css';

type ContainerProps = { children: React.ReactNode };
const Container: React.FC<ContainerProps> = ({ children }) => {
  return <div className={styles.container} data-testid='container'>{children}</div>;
};

export default Container;
