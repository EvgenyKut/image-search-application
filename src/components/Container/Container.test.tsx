import renderer from 'react-test-renderer';
import { render,screen } from '@testing-library/react';
import Container from './index';

describe('Container-component render test', () => {
  const testContent='Test'
  it('snapshot', () => {
    const tree = renderer
      .create(
        <Container>
          <></>
        </Container>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('test cases', () => {
    render(
      <Container>
        <div>{ testContent}</div>
        </Container>,
    );
    const child= screen.getByTestId('container') as HTMLInputElement;
    expect(child).toBeInTheDocument();
    expect(child).toBeVisible();
    expect(child.textContent).toBe(testContent);
  });
});
