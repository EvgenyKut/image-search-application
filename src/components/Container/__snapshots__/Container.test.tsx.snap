// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container-component render test snapshot 1`] = `
<div
  className="container"
  data-testid="container"
/>
`;
